### DISCUSSION TITLE
For cc toolchain, can we set launcher for gcc, similar as CMAKE_CXX_COMPILER_LAUNCHER in CMake?

### DISCUSSION THREAD

**Discussion by wyhao31** [bazelbuild/bazel#20094] [General]
Or is there any way to implement it using current Bazel?

**Comment by meteorcloudy** [comment]
Maybe https://stackoverflow.com/questions/41356173/how-to-use-clang-instead-g-in-bazel is relevant?

**Reply by wyhao31** [reply]
Sorry, I don't think the post is relevant to my question.

CMAKE_CXX_COMPILER_LAUNCHER is put before the actual compile command, the main reason for this in my use case is to use ccache as CMAKE_CXX_COMPILER_LAUNCHER.

**Reply by meteorcloudy** [reply]
Do you actually need ccache when using Bazel? Bazel should also have cache for all the compiling & linking actions. By setting the `CC` env var, you should be able to use ccache as the compiler. See also: https://stackoverflow.com/questions/52370202/how-to-get-bazel-ccache-and-sandboxing-to-work-together-ccache-read-only-file

**Reply by wyhao31** [reply]
We're migrating from cmake to bazel, and we use ccache in cmake setup. Besides local cache, we also use secondary cache (remote cache) for ccache, I know Bazel also has remote cache support, but we'll still use ccache in the beginning during the migration.

I don't think CC env var is an option for our use case. In CMake, CXX/CC env var is similar as CMAKE_CXX_COMPILER/CMAKE_C_COMPILER. What I want is CMAKE_CXX_COMPILER_LAUNCHER/CMAKE_C_COMPILER_LAUNCHER. I have a customized C++ toolchain, so setting CC env var is not what I want either.
